daisy/step_create_disks.go

```diff
@@ -76,3 +84,28 @@ func (c *CreateDisks) run(ctx context.Context, s *Step) DError {
 		return nil
 	}
 }
+
+// Merges two slices of features and returns a new slice instance.
+// Duplicates are removed, and the result is sorted lexically.
+func combineGuestOSFeatures(currentFeatures []*compute.GuestOsFeature,
+	additionalFeatures ...string) []*compute.GuestOsFeature {
+	featureSet := map[string]bool{}
+	for _, feature := range additionalFeatures {
+		featureSet[feature] = true
+	}
+	for _, feature := range currentFeatures {
+		featureSet[feature.Type] = true
+	}
+	ret := make([]*compute.GuestOsFeature, 0)
+	for feature := range featureSet {
+		ret = append(ret, &compute.GuestOsFeature{
+			Type: feature,
+		})
+	}
+	// Sort elements by type, lexically. This ensures
+	// stability of output ordering for tests.
+	sort.Slice(ret, func(i, j int) bool {
```

Comment:
We should not have code that is needed only for tests. If tests need sorted values, they should sort them themselves.

Reply:
Hmm. I'm not sure I agree, but it sounds like your opinion is stronger than mine, so I'm fine moving it :)

Reply:
Actually, looking at it again, I'd say I still don't agree, so let me explain more :)  Right now the resulting feature list is being tested in two places, so we'd have to re-sort it twice in our tests. Any time we do more tests, we'll have to re-sort on the test side. 

So, by having it pre-sorted, that gives a stronger contract that makes testing easier.

Reply:
This discussion is due to the fact we are returning an array when we should be returning a set. By sorting the array you're making the array behave like a set. But then we should use set in the first place. Or, leave it as is, but don't guarantee ordering and let tests verify if two arrays are equal set wise.

Reply:
Updated to keep the sorting, but remove sorting from the contract.